pkg/net/libp2p/libp2p_transport_test.go

```diff
@@ -461,6 +463,120 @@ func TestLibp2p_SendingWithTwoNodes(t *testing.T) {
 	m.testNoConnections()
 }
 
+func TestLibp2p_Messaging(t *testing.T) {
+	logger := logging.ConsoleDebugLogger
+
+	nodeA := types.NodeID("a")
+	nodeB := types.NodeID("b")
+
+	m := newMockLibp2pCommunication(t, DefaultParams(), []types.NodeID{nodeA, nodeB}, logger)
+
+	a := m.transports[nodeA]
+	require.Equal(t, nodeA, a.ownID)
+	b := m.transports[nodeB]
+	require.Equal(t, nodeB, b.ownID)
+	m.StartAll()
+
+	t.Log(">>> connecting nodes")
+
+	initialNodes := m.Membership(nodeA, nodeB)
+
+	t.Log("membership")
+	t.Log(initialNodes)
+
+	a.Connect(initialNodes)
+	b.Connect(initialNodes)
+	m.testEventuallyConnected(nodeA, nodeB)
+	m.testEventuallyConnected(nodeB, nodeA)
+
+	t.Log(">>> sending messages")
+	nodeBEventsChan := b.EventsOut()
+
+	ctx, cancel := context.WithCancel(context.Background())
+
+	var wg sync.WaitGroup
+
+	go func(ctx context.Context) {
+		wg.Add(1)
+		defer wg.Done()
+
+		<-ctx.Done()
+
+		t.Log(">>> cleaning")
+		m.StopAll()
+		m.testEventuallyNoStreamsBetween(nodeA, nodeB)
+		m.testEventuallyNoStreams(nodeA)
+		m.testEventuallyNoStreams(nodeB)
+		m.testConnsEmpty()
+
+		m.CloseHostAll()
+		m.testNoConnections()
+	}(ctx)
+
+	sent := 0
+	sentBeforeDisconnect := 0
+	received := 0
+	disconnect := make(chan struct{})
+
+	testTime := time.Duration(15)
```

Comment:
It's a bit misleading to have `testTime` in 15 "units of duration". It would be more readable to me if that was already in seconds, without the rest of the code multiplying it by seconds each time.

Reply:
Fixed